### PR TITLE
Add ITS QC module

### PR DESCRIPTION
**Pull request by JianLIUhep**
Added the ITS QC module.

### CODE REVIEW

**Comment by Barthelemy** [comment]
Thank you for bringing the ITS module to a state that allows for its review and integration in the main QC repository. 

Piotr and myself will review this PR and give you comments.

File: Modules/ITS/src/ITSRawTask.cxx [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+  else {
+    Name = Form("Occupancy/Layer%d/Stave%d/HIC%d/Layer%dStave%dHIC%dHITMAP", aLayer, aStave, aHic, aLayer, aStave, aHic);
+Title = Form("Hits on Layer %d, Stave %d, Hic %d", aLayer, aStave, aHic);
+    maxX = 7 * NColHis;
+    maxY = 2 * NRowHis;
+    nChips = 14;
+  }
+  nBinsX = maxX / SizeReduce;
+  nBinsY = maxY / SizeReduce;
+  hHicHitmap[aLayer][aStave][aHic] = new TH2I(Name, Title, nBinsX, 0, maxX, nBinsY, 0,  maxY);
```
Comment (knopers8) [inline Modules/ITS/src/ITSRawTask.cxx]: Could you make sure, that each dynamically allocated variable is later deleted during destruction of the task or other appropriate moment?

File: Modules/ITS/src/ITSRawTask.cxx [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+void ITSRawTask::getProcessStatus(int aInfoFile, int& aFileFinish)
+{
+  aFileFinish = aInfoFile % 10;
+  //cout<<"aInfoFile = "<<aInfoFile<<endl;
+  FileRest = (aInfoFile - aFileFinish) / 10;
+
+  QcInfoLogger::GetInstance() << "FileFinish = " << aFileFinish << AliceO2::InfoLogger::InfoLogger::endm;
+  QcInfoLogger::GetInstance() << "FileRest = " << FileRest << AliceO2::InfoLogger::InfoLogger::endm;
+
+  if (aFileFinish == 0)
```
Comment (knopers8) [inline Modules/ITS/src/ITSRawTask.cxx]: Please use brackets also for if statements with just one line, following the coding guidelines https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html?showone=Braces#Braces

File: Modules/ITS/src/runITS.cxx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+///
+/// QC task and Checker are instantiated by respectively TaskFactory and CheckerFactory,
+/// which use preinstalled config file, that can be found in
+/// ${QUALITYCONTROL_ROOT}/etc/qcTaskDplConfig.json or Framework/qcTaskDplConfig.json (original one).
+///
+/// To launch it, build the project, load the environment and run the executable:
+///   \code{.sh}
+///   > aliBuild build QualityControl --defaults o2
+///   > alienv enter QualityControl/latest
+///   > runTaskDPL
```
Comment (knopers8) [inline Modules/ITS/src/runITS.cxx]: Please update this piece of documentation or delete it.

File: Modules/ITS/include/ITS/ITSRawTask.h [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+#include "ITSMFTReconstruction/DigitPixelReader.h"
+
+class TH1F;
+
+using namespace o2::quality_control::core;
+
+namespace o2 {
+namespace quality_control_modules {
+namespace itsrawtask {
```
[the remark this answers is not in the document]
Comment (JianLIUhep) [reply Modules/ITS/include/ITS/ITSRawTask.h]: Changed it to "its"

File: Modules/ITS/include/ITS/ITSRawTask.h [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    TH1D *hErrorPlots;
+    TH1D *hFileNameInfo;
+    TH2D *hErrorFile;
+    TH1D *hInfoCanvas;
+        
+    TH1D *hOccupancyPlot[NLayer];
+    TH2I *hEtaPhiHitmap[NLayer];
+    TH2D *hChipStaveOccupancy[NLayer];
+    TH2I *hHicHitmap[7][48][14];
+    TH2I *hChipHitmap[7][48][14][14];
```
[the remark this answers is not in the document]
Comment (JianLIUhep) [reply Modules/ITS/include/ITS/ITSRawTask.h]: We would like to keep those for the time being.

File: Modules/ITS/include/ITS/ITSRawTask.h [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+    int TotalDigits = 0;
+    int NEvent;
+    int NEventPre;
+    int TotalFileDone;
+    //	int FileRest;
+
+    int Counted;
+    int TotalCounted = 10000;
+    int Yellowed;
```
[the remark this answers is not in the document]
Comment (JianLIUhep) [reply Modules/ITS/include/ITS/ITSRawTask.h]: The corresponding variables' names have been changed.

File: Modules/ITS/its.json [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        "location": "remote"
+      }
+    }
+  },
+  "dataSamplingPolicies": [
+    {
+      "id": "its-raw",
+      "active": "true",
+      "machines": [],
+      "dataHeaders": [
```
[the remark this answers is not in the document]
Comment (JianLIUhep) [reply Modules/ITS/its.json]: Changed to the recommended format.

File: Modules/ITS/src/ITSRawTask.cxx [hunk 8]
```diff
@@ -0,0 +1,771 @@
...
+}
+
+void ITSRawTask::initialize(o2::framework::InitContext &/*ctx*/)
+{
+  QcInfoLogger::GetInstance() << "initialize ITSRawTask" << AliceO2::InfoLogger::InfoLogger::endm;
+
+  o2::its::GeometryTGeo *geom = o2::its::GeometryTGeo::Instance();
+  geom->fillMatrixCache(o2::utils::bit2Mask(o2::TransformType::L2G));
+  int numOfChips = geom->getNumberOfChips();
+  cout << "numOfChips = " << numOfChips << endl;
```
[the remark this answers is not in the document]
Comment (JianLIUhep) [reply Modules/ITS/src/ITSRawTask.cxx]: Modifications were done.

File: Modules/ITS/src/ITSRawTask.cxx (hunk 1 above)
Comment (JianLIUhep) [reply Modules/ITS/src/ITSRawTask.cxx]: Added to the deconstructor.

File: Modules/ITS/src/ITSRawTask.cxx (hunk 2 above)
Comment (JianLIUhep) [reply Modules/ITS/src/ITSRawTask.cxx]: It's done.

File: Modules/ITS/src/runITS.cxx (hunk 3 above)
Comment (JianLIUhep) [reply Modules/ITS/src/runITS.cxx]: It's done.

File: Modules/ITS/include/ITS/ITSRawTask.h [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+#include <TBox.h>
+#include <TColor.h>
+#include <TMath.h>
+#include <TStyle.h>
+#include "Rtypes.h"		// for Digitizer::Class, Double_t, ClassDef, etc
+#include "TObject.h"		// for TObject
+#include "FairTask.h"
+#include "TPaveText.h"
+#include "TGaxis.h"
+#include "TEllipse.h"
```
[the remark this answers is not in the document]
Comment (JianLIUhep) [reply Modules/ITS/include/ITS/ITSRawTask.h]: Changed as suggested.

**Comment by knopers8** [comment]
Thanks for these fixes! I still see two errors in the CI:
```
/mnt/mesos/sandbox/sandbox/sw/SOURCES/QualityControl/build_QualityControl_o2/0/Modules/ITS/src/runITS.cxx:66:10: fatal error: QualityControl/Checker.h: No such file or directory
 #include "QualityControl/Checker.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
```
[ 91%] Building CXX object Modules/ITS/CMakeFiles/QcITSRawTask.dir/src/ITSRawTask.cxx.o
/mnt/mesos/sandbox/sandbox/sw/SOURCES/QualityControl/build_QualityControl_o2/0/Modules/ITS/src/ITSRawTask.cxx: In member function 'virtual void o2::quality_control_modules::its::ITSRawTask::monitorData(o2::framework::ProcessingContext&)':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/QualityControl/build_QualityControl_o2/0/Modules/ITS/src/ITSRawTask.cxx:229:25: error: 'class o2::itsmft::Digit' has no member named 'getROFrame'; did you mean 'getCharge'?
     mNEvent = pixeldata.getROFrame();
                         ^~~~~~~~~~
                         getCharge
```
After that, could you please apply clang-format to the project? Here it is described how to do that https://github.com/AliceO2Group/CodingGuidelines#alice-o2-coding-guidelines

**Comment by JianLIUhep** [comment]
Ok, Thank you. I went through the code and the issue is from O2 ITS rawpixelreader (changed digits format). We are working on that and then we will modify the corresponding ITS QC code.

File: Modules/ITS/include/ITS/ITSRawTask.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Modules/ITS/include/ITS/ITSRawTask.h]: You can keep them for the commissioning but it is absolutely not possible to have that number of histograms in production. Having 65856 2d histograms is not reasonable.

[the remark this answers is not in the document]
Comment (JianLIUhep) [reply Modules/ITS/include/ITS/ITSRawTask.h]: Thanks a lot for the comment. We will keep in mind and try to optimize that later.

**Comment by JianLIUhep** [comment]
It was removed in d92a6b2. Sorry I did not put the correct commit comment.

**Comment by knopers8** [comment]
You have removed there Checker.h, but not Checker.cxx

**Comment by JianLIUhep** [comment]
Sorry about that. Now it's removed

**Comment by JianLIUhep** [comment]
@knopers8 Could you please check if Travis CI got stuck? Thank you.

**Comment by Barthelemy** [comment]
@JianLIUhep We need to ask @ktf or @MichaelLettrich

**Comment by ktf** [comment]
Mmm... Looks like travis never got notified. Can you squash your commits and push again?

**Comment by JianLIUhep** [comment]
@ktf, I squashed lastest two commits but seems not to help. Shall I squash all commits of this PR? Thank you.

**Comment by knopers8** [comment]
@ktf @MichaelLettrich Travic still doesn't report any result.

**Comment by JianLIUhep** [comment]
@Barthelemy @knopers8  Shall I resubmit a new PR? It seems that Travis CI is still stuck. Than you

**Comment by knopers8** [comment]
I am not sure if this is yet in place, but it was announced yestarday, that WIP PRs won't be built. I will remove it here, maybe that will help

**Comment by Barthelemy** [comment]
Hi Jian, sorry for not reacting on this but as there was the WIP I did not think it was urgent. It seems to be still stuck. @ktf @MichaelLettrich  Could you check ? Thank you

**Comment by ktf** [comment]
For first time contributors, we require at least an APPROVED review before we test, for security reasons.

**Comment by Barthelemy** [comment]
@ktf I was not aware of this. Ok, I approved then.

**Comment by knopers8** [comment]
Me neither, is it documented somewhere?

**Comment by Barthelemy** [comment]
@ktf is travis stuck ?

**Comment by Barthelemy** [comment]
https://alice.its.cern.ch/jira/browse/O2-1267

**Comment by ktf** [comment]
I do not see the travis test for this PR at all. Can you push again?

**Comment by ktf** [comment]
@knopers8 it was announced at the WP3 meeting last week and in the computing board. I've nevertheless documented it:

**Comment by ktf** [comment]
https://github.com/AliceO2Group/AliceO2Group.github.io/blob/master/quickstart/devel.markdown#pull-request-flow

**Comment by ktf** [comment]
what we should really do is to have a message posted to the PR when a first time contributor opens a PR.

**Comment by ktf** [comment]
Could it be that this PR was done before travis was enabled for this repository? I see more recent PRs being executed just fine.

**Comment by Barthelemy** [comment]
travis has been enabled for this repo long ago I think. I will merge because we can't keep Jian waiting indefinitely. The problem remains.

**Comment by ktf** [comment]
Ok.. it's still weird because I cannot find the test at all... @awegrzyn Was it migrated to travis-ci.com?

**Comment by awegrzyn** [comment]
Yes, it was migrated to `travis-ci.com` long time ago

**Pull request merged**